### DISCUSSION TITLE
Script to import yandex music playlists to json.

### DISCUSSION THREAD

**Discussion by Nov1kov** [nukeop/nuclear#1624] [Show and tell]
I've created a Python script that allows you to export your Yandex Music playlists to JSON format for easy import into Nuclear Player.

**Supports three modes:**
- Token (recommended): Download all playlists, including deleted tracks.
- Link: Import public playlists.
- File: Import from HTML, but may not work for large playlists.

https://github.com/Nov1kov/yandex_music_to_nuclear

**Comment by nukeop** [comment]
Pretty cool. Do you think we could rewrite this in Typescript and integrate it directly with Nuclear?

**Reply by Nov1kov** [reply]
You mean adding an import button like the one in the playlist section for Spotify? That could work, but there are some limitations:

1. It will only copy a maximum of 150 tracks because there is a tricky page that loads content in parts. I haven't figured out how to get around this yet.
2. If a track has been removed from the platform by the rights holder, it will not be visible in the playlist.


This is why I personally used the unofficial API, which is not very convenient for getting a token.

But importing 150 tracks is still better than no import at all, I think.

**Reply by nukeop** [reply]
There's a similar problem to be overcome in the Spotify importer. We load the website with the playlist in an invisible frame, and keep scrolling down until we get all tracks.

**Comment by georgio893479874** [comment]
Hello

**Reply by nukeop** [reply]
General Kenobi...
